gomint-server/src/main/java/io/gomint/server/world/WorldAdapter.java

```diff
@@ -1669,7 +1900,55 @@ public int timeAsTicks() {
 
     @Override
     public Set<Entity<?>> entitiesByTag(String tag) {
+        Precondition.safeWorldAccess(this);
         return this.entityManager.findEntities(tag);
     }
 
+    public SyncSchedulerAdapter syncScheduler() {
+        return this.syncSchedulerAdapter;
+    }
+
+    @Override
+    public WorldScheduler scheduler() {
+        final PluginClassloader classLoader = CallerDetectorUtil.getCallerPluginLoader();
+        return classLoader.instance.scheduler().withWorld(this);
+    }
+
+    @Override
+    public boolean mainThread() {
+        if (this.threadId == -1) {
+            return true;
+        } else {
+            long id = Thread.currentThread().getId();
+            return this.threadId == id || this.asyncAllowedThreadId == id;
```

Comment:
Please merge the soft and hard method into one with a parameter

Reply:
done